src/mixins/canvas_events.mixin.js

```diff
@@ -834,11 +855,22 @@
      * @private
      */
     _fireEnterLeaveEvents: function(target, e) {
-      this.fireSyntheticInOutEvents(target, e, {
-        targetName: '_draggedoverTarget',
-        evtOut: 'dragleave',
-        evtIn: 'dragenter',
+      var _this = this, _draggedoverTarget = this._draggedoverTarget,
+          _hoveredTargets = this._hoveredTargets, targets = this.targets,
+          diff = _hoveredTargets.length - targets.length,
+          diffArrayLength = diff > 0 ? diff : 0,
+          diffArray = [];
+      for (var i = 0; i < diffArrayLength; i++){
+        diffArray.push(null);
+      }
```

Comment:
oh i see, the array with 3 undefined will not iterate. Well ok this is fine, if will find away to have it more terse, it can be a follow up.

![image](https://user-images.githubusercontent.com/1194048/71110090-e821ec80-21c6-11ea-8e18-5a16f4cef9bc.png)

the length is 7 but forEach will not care

Reply:
yeah forEach ignores the empties :/ interested to see what you come up with that's shorter than that. I looked at an Array.prototype.fill polyfill, but it was way more lines than a 3 for loop.